docs/contributing/contributing-code/contributing-code-tests/testing-local.md

```diff
@@ -84,3 +85,46 @@ go run ./cmd/rad/main.go workspace delete
 Another approach would be to delete and recreate the kubernetes cluster
 
 Pro tip: `kubectl config set-context --current --namespace=radius-system` to the radius-system namespace
+
+## Recipe Testing: 
+(@vishwa, @karishma)
+### Bicep Recipes: 
+### Terraform Recipes:
+- One way to create and test recipes locally is to create a a public/private git repo to host recipes and access/test it setting it's template path to the git repo location. eg:
+
+resource env 'Applications.Core/environments@2023-10-01-preview' = {
+ ...
+        recipes: {
+      'Applications.Core/extenders': {
+        default: {
+          templateKind: 'terraform'
+          templatePath:'git::https://github.com/<username>/<reponame>'
+        }
+      }
+    }
+ ...   
+
+
+- Another way to test recipes is to convert a folder containing recipes into a http server that serves that folder locally and hosts recipes.
+  ``` 
+  python command: python3 -m http.server
+  ```
+
+## CoreRP Testing:
+(@yetkin) - How to use cosmos db as a backend where one can see individual entries stored in db.
+          - Use/share Postman Collection to test api
+
+
+## Preferred Tools used to visualize k8s logs (pros/cons)
+(@yetkin/@vishwa) 
+
+
+## Dashboard/APP Graph testing
+(Bunsen Team)
+
+## Github Pipelines
+(@vinaya, @willsmith)
+
+## Container Dev
+(??) Describe good usecases to debug using dev containers
+
```

Comment:
A few usecases I really like for devcontainers

- You don't need to install anything on your machine. If you're doing a few contributions to a repo it's really nice to have an on rails process for contributing without hunting down dependencies.
- If a repo has a devcontainer then it supports codespaces. Codespaces are awesome because you can have multiple of them.
- You can run additional stateful things in a codespace/devcontainer. eg: you can have your codespace launch a Kubernetes cluster and install Radius. Then you can just close VS Code or your browser and it goes away.

This isn't really practical "do this" kind of info, but some of the reasons why devcontainers are useful.